src/proxy/main.c

```diff
@@ -0,0 +1,118 @@
+#include <errno.h>
+#include <getopt.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "libhirte/bus/bus.h"
+#include "libhirte/common/common.h"
+#include "libhirte/common/opt.h"
+#include "libhirte/log/log.h"
+
+const struct option options[] = { { ARG_HELP, no_argument, 0, ARG_HELP_SHORT }, { NULL, 0, 0, '\0' } };
+
+#define OPTIONS_STR ARG_HELP_SHORT_S
+
+static const char *opt_node_unit = NULL;
+
+static void usage(char *argv[]) {
+        printf("Usage: %s \n"
+               "\t[-h help]\t\t Print this help message.\n"
+               "\t<node>_<service>\t Input the name of the node the required service is supposed to run on.\n",
+               argv[0]);
+}
+
+int parse_node_unit_opt(const char *opt_node_unit, UNUSED char **ret_node_name, UNUSED char **ret_unit_name) {
+        if (opt_node_unit == NULL || ret_node_name == NULL || ret_unit_name == NULL) {
+                return -EINVAL;
+        }
+
+        // copy the option string as strtok_r manipulates it
+        char *rest = strdup(opt_node_unit);
+        if (rest == NULL) {
+                return -ENOMEM;
+        }
+
+        _cleanup_free_ char *token = NULL;
+        token = strtok_r(rest, "__", &rest);
+        *ret_node_name = strdup(token);
+        *ret_unit_name = strdup(strtok_r(rest, "__", &rest));
```

Comment:
Wold not hurt to do some validation of the format here, to ensure we're not returning a NULL unit name